src/StockServiceManager.php

```diff
@@ -87,9 +105,31 @@ public function listServiceIds() {
   /**
    * {@inheritdoc}
    */
-  public function getPrimaryTransactionLocation(PurchasableEntityInterface $entity, $quantity) {
+  public function getContext(PurchasableEntityInterface $entity) {
+    $store_to_use = $this->currentStore;
+    // Make sure the current store is in the entity stores.
```

Comment:
I have added code to make sure the current store is supported by the PurchasableEntity. If not it will use the first store associated with the product. This solved all the issues I have encountered so far.